src/examples/mfem_mortar_lm_patch.cpp

```diff
@@ -0,0 +1,406 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+
+/**
+ * @file mfem_mortar_lm_patch.cpp
+ *
+ * @brief Demonstrates contact patch test using the mortar method
+ *
+ * Demonstrates a three dimensional contact patch test using the mortar method
+ * in Tribol. Contact is enforced between two blocks which are initially in
+ * contact. The blocks occupy [0, 1]^3 and [0, 1]x[0, 1]x[0.99, 1.99]. To
+ * enforce symmetry and prevent rigid body modes, dirichlet boundary conditions
+ * are applied in the x-direction along the x = 0 plane, in the y-direction
+ * along y = 0 plane, and in the z-direction along the z = 0 and z = 1.99
+ * planes. Enforcement is through Lagrange multipliers and no active set (i.e.
+ * tied + sliding contact). Small deformation contact is assumed and,
+ * consequently, the system is linear and the solution is determined through a
+ * single linear solve (no timestepping).
+ *
+ * The linear system solved is
+ *  | A B^T | | d | = | f |
+ *  | B 0   | | p |   | g | ,
+ * where A is the system matrix for elasticity, B is the constraint matrix for
+ * mortar contact, d is the vector of nodal displacements, p is the vector of
+ * contact pressures, f is the vector of nodal forces, and g is the vector of
+ * gaps on the contact surface.  MFEM block operators and vectors are used to 
+ * store the linear system.
+ *
+ * The example uses the Tribol MFEM interface, which supports decomposed (MPI)
+ * meshes and will support higher order meshes (through LOR) in a future update.
```

Comment:
LOR doesn't work for implicit since the transfer of the Jacobian hasn't been implemented.  For explicit, it works.  Added a clarification as to why it isn't working for this example.